x-pack/plugins/maps/public/shared/layers/sources/wms_source/wms_client.js

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import { parseString } from 'xml2js';
+
+export class WmsClient {
+  constructor({ serviceUrl, version = '1.1.1' }) {
+    this.serviceUrl = serviceUrl;
+    this.version = version;
+  }
+
+  async _fetch(url) {
+    return window.fetch(url);
```

Comment:
Is this going to backfill OK across browsers? I doesn't look like babel is polyfiling the `window.fetch` call, and I don't think it's supported across all our supported browsers. (I tihnk just use `fetch` instead would do it).

Make this a static-method as well.

Also, do you mind adding a comment that the only reason we're putting this in the class-definition is so we can override for testing.

Reply:
Since I will be moving the WmsClient to the server, window.fetch will get replaced with node-fetch